### PR TITLE
Implement authentication

### LINKED ISSUE #7: Connect to Firebase
- [x] Set up Firebase project
- [x] Add config (make sure to load Firebase API key as JSON and add it to .gitignore)
- [x] Connect to flaks instance
- [x] Modify db.py as needed
- [x] Run tests

### LINKED ISSUE #9: Set up routes
- [x] Route to `/login`
    - [x] GET for login page
    - [x] POST to authenticate and get jwt token (figure out how to store in session)
- [x] Route to `/signup`
    - [x] GET for signup page
    - [x] POST to Firebase to create account/verify if email exists 
- [x] Route to `/status`
    - [x] GET for status page (serve tasks and resources related to this user)
    - [x] POST to add tasks

### LINKED ISSUE #11: Implement authentication
- [x] implement sign up
    - [x] check for valid email format
    - [x] re-enter password for confirmation
    - [x] redirect to log in page after successful registration
- [x] implement log in
    - [x] flash error message on same page, e.g. incorrect password, user doesn't exist
    - [x] save log in session (session-wide jwt tokens?)
- [x] check if logged in on status page (jwt tokens?)
    - [x] redirect to log in page if not logged in

### PR DESCRIPTION
**Pull request by shaynemei**
add skeleton templates to test functionality

- signup
- login
- check if logged in on status page

### CODE REVIEW

File: run_tests.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+# Check if in the right working directory.
+if [ ! -d "webapp" ]; then
+  echo "You are supposed to run this script in the Deadline root directory."
+  exit 1
+fi
+
+# Check if in Deadline virtual environment.
+source venv/deadline/bin/activate || exit 1;
```
Comment (yangevelyn) [inline run_tests.sh:10]: same question as in the other PR—does this have to match the readme?

**Review by yangevelyn — APPROVED** [review]
Tried running the code, but I'm running into issues with Firebase credentials. Pretty sure this is just an issue on my side, so LGTM!

File: run_tests.sh (hunk 1 above)
Comment (shaynemei) [reply run_tests.sh:10]: fixed readme!

**Comment by shaynemei** [comment]
> Tried running the code, but I'm running into issues with Firebase credentials. Pretty sure this is just an issue on my side, so LGTM!

We can talk about the issues and see if I can help!

**Review by labib666 — APPROVED** [review]
LGTM

**Pull request merged**